WordPress/Classes/ViewRelated/Activity/WPStyleGuide+Activity.swift

```diff
@@ -67,5 +76,48 @@ extension WPStyleGuide {
                                            lineBreakMode: .byTruncatingTail,
                                            alignment: .natural)
         }
+
+        private static func getColorByActivityStatus(_ activity: Activity) -> UIColor {
+            if activity.isStatusError {
+                return WPStyleGuide.errorRed()
+            }
+            if activity.isStatusSuccess {
+                return WPStyleGuide.validGreen()
+            }
+            if activity.isStatusWarning {
+                return WPStyleGuide.warningYellow()
+            }
+            return WPStyleGuide.greyLighten10()
+        }
+
+        // We will be able to get rid of this disgusting dictionary once we build the
+        // String->Gridicon mapping into the Gridicon module and we get a server side
+        // fix to have all the names correctly mapping.
+        private static let stringToGridiconMapping: [String: UIImage] = [
```

Comment:
😢 I'd suggest doing the mapping `[String: GridiconType]` to avoid all those `Gridicon.iconOfType`